pyomo/contrib/pynumero/interfaces/tests/test_external.py

```diff
@@ -0,0 +1,42 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+#
+import os
+import pyutilib.th as unittest
+from pyomo.contrib.pynumero.interfaces.pyomo_nlp import PyomoNLP
```

Comment:
I think you need to change this to:
```python
from pyomo.contrib.pynumero.asl import AmplInterface
if not AmplInterface.available():
    raise unittest.SkipTest(
        "Pynumero needs the ASL extension to run NLP tests")
 from pyomo.contrib.pynumero.interfaces.pyomo_nlp import PyomoNLP 
```
To guard against `pypy`, where the pynumero interface doesn't appear to work.